src/machine/machine-varlink.c

```diff
@@ -231,7 +231,7 @@ static int lookup_machine_by_pid(sd_varlink *link, Manager *manager, pid_t pid,
         if (pid <= 0)
                 return -EINVAL;
 
-        r = manager_get_machine_by_pid(manager, pid, &machine);
+        r = manager_get_machine_by_pidref(manager, &PIDREF_MAKE_FROM_PID(pid), &machine);
```

Comment:
in case pid == 0 is set, use the varlink_get_peer_pidref() to get a PidRef right-away from the peer.